### ISSUE TITLE
Documentation for Contributed Modules

### ISSUE THREAD

**Issue by Graham-72** [backdrop-ops/backdropcms.org#627]
I would like to propose that we have a new section in backdropcms.org for documentation relating to contributed modules that is user-focused rather than development focused, that is it would explain to a user what the module does, what its options are, and how to administer these when the module is installed.

It should be similar to the pages for Drupal 7 at https://www.drupal.org/docs/7/modules

It should appear in the format provided for backdropcms.org rather than present users with the different format of a github wiki.

It could include some user-friendly indexing of contrib modules particularly if there is a system of tags.

I have recently come across two situations in which this would prove helpful:
1. Having installed the module 'Responsive Favicons' I needed to refer back to notes about setting it up which is included in the module's README.md but not directly linked from my site admin here
![image](https://user-images.githubusercontent.com/10559274/71898281-d0e47800-3150-11ea-896c-e5503018d9c6.png)

2. In working on the Rules module I find it has a link to useful documentation 'Introduction to Rules' in Drupal at https://www.drupal.org/node/298480 and I would like to provide equivalent documentation within Backdrop. This screenshot show where the Rules module currently links to this documentation.
![image (9)](https://user-images.githubusercontent.com/10559274/71898803-18b7cf00-3152-11ea-8e20-c98b3280e370.png)

It would of course be voluntary to provide information about a Contributed Module though maybe the section would benefit from an editor to copy and paste suitable user-focused guidance, and I am happy to volunteer time to do this.

**Comment by olafgrabienski** [comment]
I like the idea to have documentation of contributed projects in one section of the Backdrop website. However, to maintain a separate documentation besides the individual project README and Wiki on Github would be a lot of work. As an alternative, we could re-use existing content:

As far as I know, the content of a project's Github README is already displayed on the individual project page on backdropcms.org. I guess we could re-use that content and/or the project Wiki also for a contrib documentation section.

**Comment by oadaeh** [comment]
I think both situations could point to anything anywhere. It could be a new documentation structure created on b.o, or it could be the existing b.0 or GitHub project page or the GitHub wiki (which is currently pointed to from b.o in the Documentation link), or it could be something else. It would just take modifying the module to provide that information at the appropriate place.

**Comment by oadaeh** [comment]
I will also add what I posted in Zulip:

"...As a Drupal project maintainer, keeping up on documentation has always been a sore spot for me. As a Drupal project maintainer, I am supposed to keep up to date separate documentation for each project in three areas: the README.txt file, the main project page, and that contrib documentation section. And even if they all had the same content, you still have to keep up on the three separate areas, and the formatting for at least two of them is different. For someone who does not have a lot of time for responding to and fixing issues, maintaining three separate sets of documentation is a little too much."
(Also, @BWPanda reminded me about hook_help(), a fourth area (and another format) to keep up on documentation.)

https://backdrop.zulipchat.com/#narrow/stream/218635-General/topic/Documentation.20for.20Contributed.20Modules/near/185010160

I think if we have multiple, optional places for contrib documentation, it will either be more cumbersome on the maintainer to keep them all up to date (along with normal project maintenance), or it will be more cumbersome on the end user to find the documentation being used for each project they are using.

**Comment by BWPanda** [comment]
I agree that more places for documentation might do more harm than good. Having said that, if we can copy existing documentation into different places to make it more discoverable, that'd be handy (e.g. having a new documentation section like what @Graham-72 is proposing, but pulling the content automatically from each project's GitHub wiki instead of writing/copying it manually).

**Comment by Graham-72** [comment]
I have just found something in backdropcms.org that is new to me - screenshot
![image (10)](https://user-images.githubusercontent.com/10559274/71933703-b33c0080-319a-11ea-8c58-91ef460acc95.png)

Does this mean I can add (mainly paste) documentation about the Rules module here? I would much rather add it in a Backdrop-styled page than in a github wiki. Or is 'outline' for some other purpose?

**Comment by stpaultim** [comment]
I don't think that either Drupal or Backdrop has figured out a good solution for contrib documentation yet. I support attempts at improving the situation, but I am still undecided on what the correct solution is. I'm not even sure I understand all the possibilities.

**Comment by BWPanda** [comment]
@Graham-72 'Outline' is a way of organising Book nodes I believe. Never really used it myself, but not sure it's what you think it is...

**Comment by Graham-72** [comment]
@BWPanda thanks. I have had a look around in our User Guide and can now understand what it does.

@stpaultim I have taken the liberty of adding an item 'Contributed Modules' within the User Guide (at the bottom of the index), and within that an item 'Rules'. I can start copying across to there from the Drupal 7 documentation. Then, subject to team approval, I could start on other of the contrib modules I maintain.

This is by way of an 'Aunt Sally' so we can see one possible way forward.

**Comment by BWPanda** [comment]
For posterity, here's @Graham-72's contribution/suggestion: https://backdropcms.org/user-guide/contributed-modules

I'm still against this idea, unless it's automated.

**Comment by stpaultim** [comment]
> but pulling the content automatically from each project's GitHub wiki instead of writing/copying it manually

I would agree that we don't want to maintain two versions of documentation. My argument would be that we are starting from zero (other than what is in Readme files). As far as I can see, the Github wiki pages are virtually unused.

**Comment by jenlampton** [comment]
> I'm still against this idea, unless it's automated.

I agree with @bwpanda. This model is not sustainable. 

I would expect the Rules documentation to live in its Wiki page: https://github.com/backdrop-contrib/rules/wiki

The wiki is linked from the Rules module page on backdropcms.org: https://backdropcms.org/project/rules

<img width="2456" alt="Screen Shot 2020-10-22 at 11 27 49 AM-1" src="https://user-images.githubusercontent.com/397895/96914264-9f2b6c00-1459-11eb-9d80-b843777d9e73.png">

Since the wiki link is consistent for every contrib project, I always check there first. I would not expect to find a page for contributed modules in the user guide. 

> It should appear in the format provided for backdropcms.org rather than present users with the different format of a github wiki.

@Graham-72 can you elaborate on the difference between the two formats? I'm struggling to see a difference. 

Contrib maintainers to often struggle with documentation, so I don't want to leave the responsibility and burden of creating and managing all the docs to them via README.md files.

That said, I also want to make sure this documentation is discoverable, and I expect this tiny link in the sidebar is not sufficient. 

> As far as I can see, the Github wiki pages are virtually unused.

I've seen the same thing. I think the tiny links are one of the problems.

> if we can copy existing documentation into different places to make it more discoverable, that'd be handy (e.g. having a new documentation section like what @Graham-72 is proposing, but pulling the content automatically from each project's GitHub wiki instead of writing/copying it manually).

I had exactly the same thought. But since @Graham-72 sees this information as being in different formats, I want figure out what the difference is exactly. If we pull in the text and style it to be pretty -- will that be enough?

**Comment by bugfolder** [comment]
h/t to @BWPanda for pointing out this issue from [this Zulip discussion](https://backdrop.zulipchat.com/#narrow/stream/218635-Backdrop/topic/User.20Guide.3A.20Contributed.20Modules).

Not being aware of this thread, after the Zulip discussion I'd already moved all of the User Guide Rules pages into the [Rules Wiki](https://github.com/backdrop-contrib/rules/wiki). It was not hard to bring over all the formatting intact. (I also took the liberty of doing some further Drupal-to-Backdrop conversions on some of the pages during the move, though there's still more needed.)

In the Zulip thread, I'd been in favor of keeping the documentation in one place, and now reading the arguments made there and here, I come down on the side of using the Wiki as intended.

And, N.B., I agree with the comments that the tiny link to Documentation is easy to overlook. I suspect when people see the big "Github" heading they might think "well, that's for the programmers" and wouldn't squint at the tiny type underneath it.

Ideally, if the Wiki is non-empty, then the link to it should be a great big link, at least as large as the other sidebar headings. Being documentation, it's way more important than the number of installs, so its visual prominence should reflect its importance. (And perhaps if the Wiki is empty, there's no link, so as not to lure people into going to a blank page.)

Also, since the README file is also labeled "Documentation", the Wiki link could maybe be labeled differently: "More Documentation" or "Documentation Details" or something like that.

**Comment by bugfolder** [comment]
So, thinking about possibly ways to improve the sidebar on the project page, using [Rules](https://backdropcms.org/project/rules) as an example, some thoughts. First, here's an approximation of the current sidebar:

## Current Sidebar

#### Other releases

Recommended version 1.x-2.1.5

Released: Mar 13th, 2020

[See all versions >](https://backdropcms.org/node/1134/releases)

**5,417**<br>
Downloads

**224**<br>
Active installs

### GitHub

-   [Project page](https://github.com/backdrop-contrib/rules)
-   [Issue Queue](https://github.com/backdrop-contrib/rules/issues)
-   [Documentation](https://github.com/backdrop-contrib/rules/wiki)

Clone URL

## Thoughts

* The "recommended version" isn't an "Other release." So call this section simply "Releases."
* We don't need a ">" after "See all versions". Other links don't have that.
* Call the Wiki "More Documentation", to distinguish it from the README information, which is also labeled "Documentation." And only show this link if the Wiki is non-empty. Make it big and visible.
* Change "Issue Queue" to "Report Issues" and make it a big heading. Yes, it's part of Github (as is the Wiki), but the Github-iness is not what users care about; they care about where to go when the module doesn't do what they thought.

So I suggest a sidebar that looks more like the following (subject to limitations of Markdown):

## Proposed Sidebar

### Releases

Recommended version 1.x-2.1.5

Released: Mar 13th, 2020

[See all versions](https://backdropcms.org/node/1134/releases)

#### Usage

**5,417**<br>
Downloads

**224**<br>
Active installs

### [More Documentation](https://github.com/backdrop-contrib/rules/wiki)

### [Report Issues](https://github.com/backdrop-contrib/rules/issues)

### GitHub

[Project page](https://github.com/backdrop-contrib/rules)

Clone URL

**Comment by bugfolder** [comment]
One more proposed tweak. On the [project page](https://backdropcms.org/project/rules), the first heading "DOCUMENTATION" isn't needed (and in fact, that first section really isn't the "documentation" for the module, it's more of an "introduction"). So I suggest dropping that heading from the project page. And then in the right sidebar, the link to the Wiki can just be called "Documentation."

**Comment by bugfolder** [comment]
...and the "Documentation" link to the Wiki should only be shown if there's actually a Wiki page in existence. If there's not one, then following the Wiki link just takes the person to the project page, which shows the README file all over again—so nothing new.

**Comment by bugfolder** [comment]
Thanks for those suggestions! "View or Report Issues" covers both options, makes sense as a link to the issues queue, and doesn't require additional verbiage for explanation. (Putting "View" first is a slight nudge for people to look at issues before creating new ones.)

**Comment by oadaeh** [comment]
I'm coming a bit late to this game, but what does "issue" mean to someone who is not in the know, especially if it's the only word? I think to be clear to people who are not familiar with the development workflow, you're going to have to use something like "Report a problem/request a feature."
